src/conversation.js

```
@@ -518,31 +530,45 @@ class Conversation extends Syncable {
 
 
   /**
-   * Delete the conversation from the server.
+   * Delete the Conversation from the server.
+   *
+   * This call will support various deletion modes.  Calling without a deletion mode is deprecated.
+   *
+   * Deletion Modes:
+   *
+   * * layer.Constants.DELETION_MODE.ALL: This deletes the local copy immediately, and attempts to also
+   *   delete the server's copy.
    *
    * Executes as follows:
+   *
    * 1. Submits a request to be sent to the server to delete the server's object
    * 2. Delete's the local object
    * 3. If there is an error, no errors are fired except by layer.SyncManager, but the Conversation will be reloaded from the server,
    *    triggering a conversations:add event.
    *
-   *
    * @method delete
-   * @param {boolean} destroy - if true, delete for all users, else just for this user. (false not yet supported by server)
+   * @param {number} deletionMode - layer.Constants.DELETION_MODE.ALL is only supported mode at this time
    * @return null
    */
-  delete(destroy) {
+  delete(mode) {
     const id = this.id;
-    if (!this.isTempId()) {
-      const client = this.getClient();
-      this._xhr({
-        method: 'DELETE',
-        url: '?destroy=' + Boolean(destroy),
-      }, result => {
-        if (!result.success) Conversation.load(id, client);
-      });
+    const modeValue = 'true';
+    if (mode === true) {
```

Also if `false` in not supported should we throw an error here?